### DISCUSSION TITLE
Bluetooth connection hangs / too many messages

### DISCUSSION THREAD

**Discussion by Kermit647** [pybricks/support#1081] [General]
If you spam too many print() in the code in short time, it will make the connection hang.

Result is the programm is reportet running by the ide and cant be stopped.
After closing and restarting the ide, it is not possible to connect with bluetooth again. The computer has to be restartet, so perhaps somthing in the driver hangs.
There should perhaps be some mechanism to limit the amount of messages send to the ide with print()

This is a very low priority problem and can easy be avoided by yourself by limiting the amount of print data, just wanted top report it if others have a problem with the bluetooth connection  to look for this.

**Comment by laurensvalk** [comment]
Can you share a small example program that reproduces this?

**Reply by Kermit647** [reply]
A simple

```python
while True:
    print("Hello World")
```

will do :)

**Reply by laurensvalk** [reply]
Which hub? Which Pybricks version? (see bottom right of your screen)

I cannot reproduce this on the Prime Hub with Pybricks v3.3.0b5.

**Reply by Kermit647** [reply]
Sorry, I am using the current beta with a Prime Hub, pc is a windows 11 laptop.

**Comment by dlech** [comment]
> After closing and restarting the ide, it is not possible to connect with bluetooth again. The computer has to be restartet, so perhaps somthing in the driver hangs.

This does sounds like a problem with the Bluetooth adapter and/or driver. Can you share your Bluetooth radio information? Run the following in PowerShell:

    wget https://github.com/Microsoft/busiotools/raw/master/bluetooth/tracing/GetBluetoothRadioInfo.ps1 -UseBasicParsing | iex

**Reply by dlech** [reply]
You could also try logging Bluetooth packets to see if there is anything unusual there.

https://bleak.readthedocs.io/en/latest/troubleshooting.html#windows-10

**Comment by Kermit647** [comment]
I just wanted to investigate further, however I installed the newest beta firmware update and the problem seems to be gone ! 

I dont know if this is a random coincidence, but anyway thank you for your great work !

**Reply by dlech** [reply]
We did make some fixes to Bluetooth, so perhaps not a coincidence.

**Reply by dlech** [reply]
Also see #1095 which has a reproducible case where the Bluetooth chip crashes causing a hang on City and Technic hubs.
